doc/getting-started/building-from-source.rst

```diff
@@ -3,6 +3,39 @@
 Build ACRN from Source
 ######################
 
+Introduction
+************
+
+ACRN hypervisor does not support one binary for all platforms.
+
+Hypervisor binary is generated based on the platform configuration.
+Different platforms with different configurations shall be set up explicitly
+via Kconfig.
+Instruction can be found in :ref:`getting-started-hypervisor-configuration`.
+
+One binary for all platforms is not supported because dynamic parsing is
+restricted in ACRN hypervisor, for the following considerations:
+
+* **Meet functional safety requirements** Absence of dynamic objects is
+  required in functional safety standards. Implementation of the dynamic parsing
+  would introduce the dynamic objects more or less. Restricting the dynamic
+  parsing would help ACRN hypervisor meet functional safety requirements.
+
+* **Reduce complexity** ACRN is a lightweight reference hypervisor, built for
+  embedded IoT. Since the platforms for embedded system is changing frequently,
+  support of one binary for all platforms will become more and more complex.
+  Restricting the dynamic parsing would help reduce the complexity of the
+  hypervisor.
+
+* **Keep small footprint** Implementation of the dynamic parsing would introduce
+  hundreds or thousands of code. Restricting the dynamic parsing would help keep
+  Lines of Code (LOC) of the hypervisor in a desirable range (around 30K).
+
+* **Improve boot up time** Dynamic parsing at run time would increase the boot
+  up time. Restricting the dynamic parsing would help improve boot up time of
+  the hypervisor.
+
+
```

Comment:
In the meantime, to help developers to try ACRN easily in the might-be-diversity platform, one can use the image built for generic platform, which will likely work for most platforms.